packages/eas-cli/src/commands/update/republish.ts

```diff
@@ -0,0 +1,373 @@
+    const flags = this.sanitizeFlags(rawFlags);
+
+    const {
+      projectConfig: { exp, projectId },
+      loggedIn: { graphqlClient },
+    } = await this.getContextAsync(UpdateRepublish, {
+      nonInteractive: flags.nonInteractive,
+    });
+
+    if (flags.json) {
+      enableJsonOutput();
+    }
+
+    const existingUpdates = await getOrAskUpdatesAsync(graphqlClient, projectId, flags);
+    const updatesToPublish = existingUpdates.filter(update =>
+      flags.platform.includes(update.platform as Platform)
+    );
+
+    if (existingUpdates.length === 0) {
+      throw new Error(`There are no published updates found`);
+    }
+    if (updatesToPublish.length === 0) {
+      throw new Error(
+        `There are no updates on branch "${
+          existingUpdates[0].branchName
+        }" published for the platform(s) "${rawFlags.platform}" with group ID "${
+          flags.groupId ? flags.groupId : updatesToPublish[0].groupId
+        }". Did you mean to publish a new update instead?`
+      );
+    }
+
+    if (rawFlags.platform === 'all') {
+      Log.withTick(`The republished update will appear only on: ${rawFlags.platform}`);
+    } else {
+      const platformsFromUpdates = updatesToPublish.map(update => update.platform);
+      if (platformsFromUpdates.length < defaultRepublishPlatforms.length) {
+        Log.warn(`You are republishing an update that wasn't published for all platforms.`);
+      }
+
+      Log.withTick(
+        `The republished update will appear on the same platforms it was originally published on: ${platformsFromUpdates.join(
+          ', '
+        )}`
+      );
+    }
+
+    // This command only republishes a single update group
+    // The update group properties are the same for all updates
+    const groupId = updatesToPublish[0].groupId;
+    const runtimeVersion = updatesToPublish[0].runtimeVersion;
+    const branchName = flags.branchName ?? updatesToPublish[0].branchName;
+
+    // Prevent users from republishing updates to a different branch.
+    // When using environment variables, the branch name is not updated, possibly causing unexpected sideeffects.
+    assertBranchNameIsEqualToExistingUpdateBranch(updatesToPublish, branchName, flags);
+
+    // If codesigning was created for the original update, we need to add it to the republish
+    const codeSigningByPlatform = await PublishQuery.getCodeSigningInfoFromUpdateGroupAsync(
+      graphqlClient,
+      groupId
+    );
+    const shouldCodeSignRepublish = Object.keys(codeSigningByPlatform).length > 0;
+    if (shouldCodeSignRepublish) {
+      Log.withTick(
+        `The republished update will be signed with the same codesigning as the original update.`
+      );
+    }
+
+    const updateMessage = await getOrAskUpdateMessageAsync(updatesToPublish, flags);
+    const { branchId } = await ensureBranchExistsAsync(graphqlClient, {
+      appId: projectId,
+      branchName,
+    });
```

Comment:
```suggestion
```
can just query `Update.branchId` in `getOrAskUpdatesAsync`

Reply:
Yeah, this was related to https://github.com/expo/eas-cli/pull/1533#discussion_r1030463645, specifically:

> At first, I made it possible to use both `--group` and `--branch` to "promote" and the existing update group to a different branch. But after talking with @jonsamp, we think this is a foot gun. ...

Reply:
Done!